include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -223,108 +218,155 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 					std::array<uint64_t, 4> hash = {};
 					// If true, then `getIncludeStandard` was used to find, otherwise `getIncludeRelative`
 					bool standardInclude = false;
-					nbl::system::IFileBase::time_point_t lastWriteTime = {};
 				};
 
 				struct SMacroData {
 					std::string identifier;
 					std::string definition;
+					NLOHMANN_DEFINE_TYPE_INTRUSIVE(SMacroData, identifier, definition);
 				};
 
-				struct SPreprocessorData {
+				struct SCompilerArgs; // Forward declaration for SPreprocessorArgs's friend declaration
+
+				struct SPreprocessorArgs final {
 
-					inline bool operator==(const SPreprocessorData& other) const {
+					friend class SCompilerArgs;
+					friend class SEntry;
+
+					// Public default needed for json serialization
+					SPreprocessorArgs() {};
```

Comment:
you can make the json serializaiton functions friends

Reply:
This seems reasonable